### ISSUE TITLE
Make Accessor formatting differently than method formatting possible.

### ISSUE THREAD

**Issue by mkrueger** [dotnet/roslyn#84]
Currently method and accessor formatting is the same. 
However there are many styles (mono/sharpdevelop) where these are different

```
public int Foo {
   get {
       return foo;
   }
}

public int Bar()
{
   return bar;
}
```

should be possible. 

I suggest adding a NewLinesForBracesInAccessors option to handle all the accessor cases.

**Comment by DustinCampbell** [comment]
I love it.

**Comment by DustinCampbell** [comment]
@mkrueger -- are there other formatting options we might want to add?

**Comment by tmat** [comment]
Also when overriding or otherwise auto-generating property getter we should consider "expression body" style.

**Comment by DustinCampbell** [comment]
@tmat, agreed. Though that should be optional behavior.

**Comment by tmat** [comment]
Perhaps it should be a refactoring to switch between expression bodied and block bodied members.

**Comment by DustinCampbell** [comment]
I think that's orthogonal to an option for code generation, but yes, that would help round out the support. FWIW, the code is extremely easy to write.

https://github.com/DustinCampbell/CSharpEssentials/tree/master/Source/CSharpEssentials/UseExpressionBodiedMember

**Comment by tmat** [comment]
Somewhat orthogonal. I wouldn't need to deal with an option if I had a quick fix. Let the IDE generate whatever and if I don't like it I switch it. Like with inline rename - we guess a name and if the user doesn't like it they can rename.

**Comment by DustinCampbell** [comment]
Suppose there wasn't an option and we generated it in the opposite way than you wanted? Then, you would *always* have to use the refactoring and you would eventually show up at my office door asking for an option. :smile:

Let's just have both Tomas.

**Comment by mkrueger** [comment]
@DustinCampbell I'll need to check the rest of our 'weird' formatting :). But for what I've seen so far that one is the only thing missing in supporting it.

The old NR5 engine had a really extensive set of options and was a bit too fine grained.

What I find a bit limiting is that the brace style is a bool instead of an enumeration. My goal for an formatting engine would be to support all the indent styles in:

http://en.wikipedia.org/wiki/Indent_style

For doing that a bool wouldn't be enough for defining brace styles for a construct. In NR5 it was possible to set different brace styles for almost every construct :). But as said maybe a bit too fine grained and atm I'm just happy if the mono style works - but maybe a consideration for making the formatting engine more flexible.

**Comment by tmat** [comment]
Having an option is certainly good :) I'm just saying having a quick fix goes a long way.

**Comment by Pilchie** [comment]
Given our focus on shutting down for VS 2015, I don't think we can take this now.  @mkrueger: I think you said you had a workaround for now?  Also, presumably if we fix it shortly after VS2015, you could leverage an out-of-band release of our NuGet packages?

**Comment by mkrueger** [comment]
No there is no workaround - I think we need a custom build.
We had several issues with the formatter btw.

**Comment by Pilchie** [comment]
Fixed by #5535
